### PR TITLE
Position keep-to-library widget so that it's always visible.

### PR DESCRIPTION
**Pull request by yipingliao**
Also, add more customizable up/down offsets.
Highlight tag when a widget is open for that tag.

See: https://app.asana.com/0/18079737520750/18785877576576

@andrewconner cc @ahsu1230

### CODE REVIEW

File: kifi-backend/modules/shoebox/angular/src/keep/keepToLibraryWidget.js [hunk 1]
```diff
@@ -19,7 +19,13 @@ angular.module('kifi')
        *   keptToLibraries - an array of library ids that are already keeping the keep.
        *   clickAction() - a function that can be called once a library is selected;
        *                   called with the element that this widget is on.
-       *   libSelectTopOffset - amount to shift up relative to the element that has this directive as an attribute.
+       *   exitAction() - a function that is called once the widget exits (includes when the widget exits after an action).
+       *
+       *   Positioning properties:
+       *   ----------------------
+       *   libSelectDownOffset - shift the bottom of the widget this much below the element.
```
Comment (yipingliao) [inline kifi-backend/modules/shoebox/angular/src/keep/keepToLibraryWidget.js:26]: Look here first: this explains what the positioning properties are.

File: kifi-backend/modules/shoebox/angular/src/keep/keepToLibraryWidget.js [hunk 2]
```diff
@@ -139,25 +149,64 @@ angular.module('kifi')
...
+            var left = element.offset().left - shiftLeftDistance;
+
+            // Shift the widget up or down.
+            var shiftUpDistance ;
             var widgetHeight = widget.height();
 
-            // If there is enough room at the bottom, drop the widget down.
-            // Otherwise, scoot up just enough to fit the widget.
-            var widgetOffset = (distanceFromBottom - widgetHeight > 0) ? 0 : widgetHeight - distanceFromBottom;
+            // First, shift the widget down based on the passed in desired shift down distance.
```
Comment (yipingliao) [inline kifi-backend/modules/shoebox/angular/src/keep/keepToLibraryWidget.js:186]: Look here second: these comments explain how the widget is positioned up and down.

File: kifi-backend/modules/shoebox/angular/src/tagManage/tagManage.js [hunk 3]
```diff
@@ -162,10 +163,15 @@ angular.module('kifi')
     $scope.actionToLibrary = '';
 
     $scope.changeSelection = function (tag, action) {
+      tag.selected = true;
```
Comment (yipingliao) [inline kifi-backend/modules/shoebox/angular/src/tagManage/tagManage.js:166]: @ahsu1230 How I am doing highlighting in tags. Let me know if you have a better method!

File: kifi-backend/modules/shoebox/angular/src/tagManage/tagManage.js [hunk 4]
```diff
@@ -162,10 +163,15 @@ angular.module('kifi')
...
       $scope.selectedTag = tag;
       $scope.actionToLibrary = action;
     };
 
+    $scope.exitAction = function () {
```
Comment (yipingliao) [inline kifi-backend/modules/shoebox/angular/src/tagManage/tagManage.js:171]: @ahsu1230 The widget now looks for an `exitAction` and invokes it whenever it exits.

File: kifi-backend/modules/shoebox/angular/src/tagManage/tagManage.tpl.html [hunk 5]
```diff
@@ -24,7 +24,7 @@
           <ul class="manage-tag-list">
             <li class="manage-tag-row" ng-repeat="tag in tagsToShow">
               <div class="manage-tag-row-name-section">
-                <a class="manage-tag-row-name" title="{{tag.name}}" ng-href="{{tag.href}}">{{tag.name}}</a>
+                <a class="manage-tag-row-name" title="{{tag.name}}" ng-href="{{tag.href}}" ng-class="{'selected-tag': tag.selected}">{{tag.name}}</a>
```
Comment (ahsu1230) [inline kifi-backend/modules/shoebox/angular/src/tagManage/tagManage.tpl.html:27]: @yipingliao I would do it this way too! haha good job =)

**Comment by ahsu1230** [comment]
:+1: you have a merge conflict! =)

**Pull request closed without merging**